Feature/{chimereucheya}/Article page

Merdoth opened the pull request:
#### What does this PR do?
-Application Article page

#### Description of Task to be completed?
- Shows the application article page which gives full insight into the type of articles published by lifeAid.

#### How should this be manually tested?
- run the application using `npm run dev` visit `http://localhost:3000/article`

#### Any background context you want to provide?
N/A

#### What are the relevant pivotal tracker stories?
- story type: Feature
- story title: Article page

#### Screenshots (if appropriate)
N/A

#### Question
N/A

Commit:
Add article page file
Add article page css file
Add article page display screen
Fix article page css file content indentation

Commit:
Merge branch 'feature/{chimereucheya}/servicePage' of https://github.com/lifeaid/project-template into feature/{chimereucheya}/articlePage
Resolve merge conflict

Commit:
fix application design placement errors

enyeInc commented on components/Article/ArticlePage.js:
```diff
@@ -0,0 +1,89 @@
+                            </a>
+                        </p>
+                    </div>
+                </div>
```
content here can be mapped

enyeInc commented on components/Article/ArticlePage.js:
```diff
@@ -0,0 +1,89 @@
+                            </a>
+                        </p>
+                    </div>
+                </div>
```
all of this content can created through an array mapping

Merdoth replied:
Will do that once I start connecting it to the api directly

enyeInc commented on components/Article/ArticlePage.js:
```diff
@@ -0,0 +1,89 @@
+                        </p>
+                    </div>
+                </div>
+                <p className="pagination"> 1  |  2  |  3  |  4  |  ...</p>
```
use antd list component to handle this list of contents and the pagination logic

Merdoth replied:
Done

Merdoth replied:
Will do that once I start connecting it to the api directly

enyeInc commented on components/Service/ServicePage.js:
```diff
@@ -0,0 +1,172 @@
+                        </div>
+                    </div>
+                </div>
+            </div>
```
map over all of these contents

Merdoth replied:
Will do that once I start connecting it to the api directly

Way2nnadi replied:
Do that now in this PR @Merdoth, practice writing cleaner up.

enyeInc commented on components/Service/ServicePage.js:
```diff
@@ -0,0 +1,172 @@
+                        </span>
+                    </div>
+                </div>
+            </div>
```
map over these contents

Merdoth replied:
Will do that once I start connecting it to the api directly

enyeInc commented:
@Merdoth Your code overall looks good, you just have to adopt the concept of DRY (Don't Repeat Yourself). If you find yourself writing what looks like the same code more than one, you need to abstract that out into a function or a data structure of some kind

Commit:
Add report section to application
Add new css design for the report section and made adjustments to the old designs
Add antd design package to the application inorder to have access to the pagination component

EngrEric commented on components/Article/ArticlePage.js:
```diff
@@ -0,0 +1,118 @@
+import { Pagination } from 'antd';
+import NavigationBar from '../Common/NavigationBar';
+import { REPORT } from './constants';
+import { HOME_STRINGS } from '../constants';
```
keep all the constants for this component in the same place `import { HOME_STRINGS, REPORT, SERVICE_STRING  } from './constants';`

EngrEric commented on components/Article/ArticlePage.js:
```diff
@@ -0,0 +1,118 @@
+
+const {
+    REPORT_TEXT,
+} = REPORT;
```
`const { REPORT_TEXT } = REPORT;`

EngrEric commented on components/Article/ArticlePage.js:
```diff
@@ -0,0 +1,118 @@
+
+const {
+    GOOGLE_FONTS,
+} = SERVICE_STRING;
```
`const { GOOGLE_FONTS } = SERVICE_STRING;`

EngrEric commented on components/Article/ArticlePage.js:
```diff
@@ -0,0 +1,118 @@
+const {
+    ARTICLE_READ,
+    READ_MORE,
+} = HOME_STRINGS;
```
`const { ARTICLE_READ, READ_MORE } = HOME_STRINGS;`

EngrEric commented on components/Article/ArticlePage.js:
```diff
@@ -0,0 +1,118 @@
+                    </p>
+                </div>
+                <div className="article-content">
+                    <p className="article-content-header">What Should be in a First Aid kit?</p>
```
move all strings to constants and map over all these. always practice DRY (do not repeat yourself) when possible.

Merdoth replied:
Those are temporary headers that will be written

EngrEric commented on components/Service/ServicePage.js:
```diff
@@ -0,0 +1,172 @@
+import NavigationBar from '../Common/NavigationBar';
+import './service.scss';
+import { HOME_STRINGS } from '../constants';
+import { SERVICE_STRING } from './constants';
```
merge these constants to same place

Merdoth replied:
Like you said I should practice DRY so I can't be repeating the same string that already exists when I can just import it from the initial place and if you pay enough attention to details you'll notice that those two constants are serving two very different purpose

EngrEric reviewed (changes requested):
clean up your code and also resolve the conflicts

Commit:
fix merge conflict

Commit:
Merge branch 'development' of https://github.com/lifeaid/project-template into feature/{chimereucheya}/articlePage
fix merge conflict
